OpenAssetIO/OpenAssetIO
# Add `kManagerDriven` and `kRequired` access modes

## Description

Closes #1209

Add the `kRequired` and `kManagerDriven` access modes for use with `managementPolicy`; and rename `ResolveAccess.kWrite` to `kManagerDriven`, for consistency.


## Test Instructions

See
- https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/76
- https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/99
